octopusdeploy_framework/resource_lifecycle.go

```diff
@@ -125,13 +131,49 @@ func (r *lifecycleTypeResource) Update(ctx context.Context, req resource.UpdateR
 		return
 	}
 
+	handleUnitCasing(updatedLifecycle, lifecycle)
+
 	data = flattenLifecycleResource(updatedLifecycle)
 
 	removeDefaultRetentionPolicy(releaseRetentionPolicySet, data, defaultPolicy, tentacleRetentionPolicySet)
 
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
+func handleUnitCasing(resource *lifecycles.Lifecycle, data *lifecycles.Lifecycle) {
+	// Set state to the casing provided in the desired state, as the Api will always return capitalised units
+	resource.ReleaseRetentionPolicy = updateRetentionPeriodUnit(resource.ReleaseRetentionPolicy, data.ReleaseRetentionPolicy.Unit)
+	resource.TentacleRetentionPolicy = updateRetentionPeriodUnit(resource.TentacleRetentionPolicy, data.TentacleRetentionPolicy.Unit)
+
+	if len(data.Phases) == 0 {
+		return
+	}
+
+	for i, phase := range resource.Phases {
+		if phase.ReleaseRetentionPolicy != nil && len(phase.ReleaseRetentionPolicy.Unit) != 0 {
```

Comment:
Is there a reason why we don't just check if the string is empty or not?

Reply:
Yeah, since phase policies can be nil, if `phase.ReleaseRetentionPolicy` is `nil` and it tries to access `phase.ReleaseRetentionPolicy.Unit` it throws a nil pointer dereference error

Reply:
Oh my bad, no reason but I can change it if the other way is preferred

Reply:
Both are technically fine - but to me, comparing empty string has a more clear intent. I'm not attached to it in any way though